pages/2_DataRAG.py

```
@@ -11,7 +13,59 @@
+# Check if the DataFrame is available in session state
+if "df" in st.session_state:
+    df = st.session_state["df"]
+
+    # Ensure the STATUS column exists
+    if 'STATUS' in df.columns:
+        editable_column = "STATUS"  # Specify the column you want to make editable
+        non_editable_columns = [col for col in df.columns if col != "STATUS"]
+        allowed_values = ["R", "A", "G"]  # Dropdown values
+
+        # Streamlit editable table with dropdown for the STATUS column
+        edited_df = st.data_editor(
```

Ruff (0.8.2)

`st` may be undefined, or defined from star imports

(F405)